### PR TITLE
[ZMQ] append a message sequence number to every ZMQ notification

### PR DESCRIPTION
**Pull request by chaeplin**
Code from bitcoin main branch.
- Fixes ZMQ startup with bad arguments(7621)
- [ZMQ] append a message sequence number to every ZMQ notification(7762)

contrib/zmq/zmq_sub.py to python 3 compatible

tested / sample log https://gist.github.com/chaeplin/cdeb14a8a5fcc7974109af30696224d0

### CODE REVIEW

**Comment by chaeplin** [comment]
travis-ci not passed

**Comment by UdjinM6** [comment]
Yeah, this still happens sometimes for 32bit build. Failed at "-salvagewallet" this time which shouldn't be affected by changes here so I restarted the job. Feel free to reopen this if it passes.

**Comment by schinzelh** [comment]
Yeah, don't get dragged down by Travis - sometimes it just has a bad day :-)

File: doc/zmq.md [hunk 1]
```diff
@@ -101,3 +101,8 @@ using other means such as firewalling.
 Note that when the block chain tip changes, a reorganisation may occur
 and just the tip will be notified. It is up to the subscriber to
 retrieve the chain from the last known block to the new tip.
+
+There are several possibilities that ZMQ notification can get lost
+during transmission depending on the communication type your are
+using. Bitcoind appends an up-counting sequence number to each
```
Comment (UdjinM6) [inline doc/zmq.md]: s/Bitcoind/Dashd/

**Pull request merged**